File: langs/ruby/problems/gpu_scheduler/stub.rb
class Simulation
  def initialize
  end
  def add_gpu(gpu_id, mem)
    raise 'not implemented'
  end
  def submit_job(job_id, mem)
    raise 'not implemented'
  end
  def status(job_id)
    raise 'not implemented'
  end
  def assign()
    raise 'not implemented'
  end
  def complete(job_id)
    raise 'not implemented'
  end
  def cancel(job_id)
    raise 'not implemented'
  end
  def set_priority(job_id, priority)
    raise 'not implemented'
  end
end

File: langs/typescript/problems/gpu_scheduler/stub.ts
class Simulation {
  constructor() {}
  addGpu(gpuId, mem) { throw new Error('not implemented'); }
  submitJob(jobId, mem) { throw new Error('not implemented'); }
  status(jobId) { throw new Error('not implemented'); }
  assign() { throw new Error('not implemented'); }
  complete(jobId) { throw new Error('not implemented'); }
  cancel(jobId) { throw new Error('not implemented'); }
  setPriority(jobId, priority) { throw new Error('not implemented'); }
}
module.exports = { Simulation };
